### PR TITLE
Integrate Numbers Protocol

### PR DESCRIPTION
**Pull request by nwingt**

### CODE REVIEW

File: components/CheckBox.vue [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      <IconCheckBoxOff
+        v-else
+        :class="iconClass"
+      />
+      <slot />
+    </label>
+  </span>
+</template>
+
+<script>
```
Comment (AuroraHuang22) [inline components/CheckBox.vue:35]: Not using TypeScript?

File: server/arweave/router.ts [hunk 2]
```diff
@@ -81,9 +59,17 @@ router.post('/upload',
 
     // shortcut for existing file without checking tx
     if (existingArweaveId) {
+      let numAssetsIds = [];
+      if (req.body.num === '1') {
```
Comment (williamchong) [inline server/arweave/router.ts:63]: Would `req.body.num && req.body.num !== '0'` be more robust?

File: components/CheckBox.vue (hunk 1 above)
Comment (williamchong) [inline components/CheckBox.vue:35]:
```suggestion
<script lang="ts">
```

**Review by williamchong — CHANGES_REQUESTED** [review]

File: components/CheckBox.vue (hunk 1 above)
Comment (nwingt) [reply components/CheckBox.vue]: Updated

File: server/arweave/router.ts [hunk 3]
```diff
@@ -115,11 +101,25 @@ router.post('/upload',
...
       return;
     }
-    const [{ arweaveId, list }] = await Promise.all([
+    const promises: Promise<any>[] = [
       uploadFilesToArweave(arFiles),
       uploadFilesToIPFS(arFiles),
-    ]);
-    res.json({ arweaveId, ipfsHash, list });
+    ];
+    if (req.body.num === '1') {
```
[the remark this answers is not in the document]
Comment (nwingt) [reply server/arweave/router.ts]: Updated

File: server/arweave/router.ts (hunk 2 above)
Comment (nwingt) [reply server/arweave/router.ts]: Updated

File: components/CheckBox.vue (hunk 1 above)
Comment (nwingt) [reply components/CheckBox.vue]: Updated

**Review by williamchong — APPROVED** [review]

**Pull request merged**